addons/mail/static/src/messaging/component/thread_needaction_preview/thread_needaction_preview.js

```diff
@@ -0,0 +1,120 @@
+
+class ThreadNeedactionPreview extends Component {
+
+    /**
+     * @override
+     */
+    constructor(...args) {
+        super(...args);
+        useStore(props => {
+            const thread = this.env.entities.Thread.get(props.threadLocalId);
+            const mainThreadCache = thread ? thread.mainCache : undefined;
+            let lastNeedactionMessageAuthor;
+            let lastNeedactionMessage;
+            let threadCorrespondent;
+            if (thread) {
+                lastNeedactionMessage = mainThreadCache.lastNeedactionMessage;
+                threadCorrespondent = thread.correspondent;
+            }
+            if (lastNeedactionMessage) {
+                lastNeedactionMessageAuthor = lastNeedactionMessage.author;
+            }
+            return {
+                isDeviceMobile: this.env.messaging.device.isMobile,
+                lastNeedactionMessage: lastNeedactionMessage ? lastNeedactionMessage.__state : undefined,
+                lastNeedactionMessageAuthor: lastNeedactionMessageAuthor
+                    ? lastNeedactionMessageAuthor.__state
+                    : undefined,
+                thread: thread ? thread.__state : undefined,
+                threadCorrespondent: threadCorrespondent
+                    ? threadCorrespondent.__state
+                    : undefined,
+            };
+        });
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Get the image route of the thread.
+     *
+     * @returns {string}
+     */
+    image() {
+        if (this.thread.moduleIcon) {
+            return this.thread.moduleIcon;
+        }
+        if (this.thread.correspondent) {
+            return `/web/image/res.partner/${this.thread.correspondent.id}/image_128`;
+        }
+        if (this.thread.model === 'mail.channel') {
+            return `/web/image/mail.channel/${this.thread.id}/image_128`;
+        }
+        return '/mail/static/src/img/smiley/avatar.jpg';
+    }
+
+    /**
+     * Get inline content of the last message of this conversation.
+     *
+     * @returns {string}
+     */
+    get inlineLastNeedactionMessageBody() {
+        if (!this.thread.lastNeedactionMessage) {
+            return '';
+        }
+        return mailUtils.parseAndTransform(this.thread.lastNeedactionMessage.prettyBody, mailUtils.inline);
+    }
+
+    /**
+     * @returns {mail.messaging.entity.Thread}
+     */
+    get thread() {
+        return this.env.entities.Thread.get(this.props.threadLocalId);
+    }
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClick(ev) {
+        this.trigger('o-select-thread', { thread: this.thread.localId });
```

Comment:
Didn't we allow ourselves to send entities with trigger?

Reply:
Indeed, made change also in thread preview